cpp/src/plasma/test/client_tests.cc

```diff
@@ -645,15 +644,36 @@ TEST_F(TestPlasmaStore, RepeatlyCreateGPUTest) {
     std::shared_ptr<Buffer> data;
     ARROW_CHECK_OK(client_.Create(object_id, data_size, 0, 0, &data, 1));
     ARROW_CHECK_OK(client_.Seal(object_id));
-
-    data = nullptr;
     ARROW_CHECK_OK(client_.Release(object_id));
   }
 
   // delete all
   ARROW_CHECK_OK(client_.Delete(object_ids));
 }
 
+TEST_F(TestPlasmaStore, LetBufferAutoRelease) {
+  ObjectID object_id = random_object_id();
+  const int64_t data_size = 40;
+
+  std::shared_ptr<Buffer> create_buff;
+  ARROW_CHECK_OK(client_.Create(object_id, data_size, nullptr, 0, &create_buff, 1));
+  ARROW_CHECK_OK(client_.Seal(object_id));
+  ARROW_CHECK_OK(client_.Release(object_id));
```

Comment:
create_buff is created from a GpuProcessHandle->ptr.
In the original version, "create_buff = nullptr;" should be added here to let the process run OK.